internal/fixture/state.go

```diff
@@ -364,6 +389,9 @@ type TestCase struct {
 	// Object, if not empty, is the object as it's expected to
 	// be after all the operations are run.
 	Object typed.YAMLObject
+	// APIVersion should be set if the object is non-empty and
```

Comment:
The diff would be smaller if this defaulted to v1 or something.

I think the comment should explain that for the tests, we implemented this as selecting the root object in the schema? Some comment should at least explain that.

Reply:
I'm really not a big fan of using a default. I considered maybe "default to the version of the latest change", which is what we implicitly used to do.

> I think the comment should explain that for the tests, we implemented this as selecting the root object in the schema? Some comment should at least explain that.

Yeah, I agree.

Reply:
I think an explicit hard coded default is less magical than going with the prior thing.

If you want to test multi-version stuff, you should specify this with every change for consistency. If you don't want to test multi version stuff, you don't really care what this defaults to, no?

Reply:
Yeah, I really dislike the default, it really looks much more magic. I'm also fine with just specifying the version everywhere to be honest, I don't think it hurts that much.